File: legacy/leet-11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        j = len(height) - 1
        i = 0
        ans = 0
        while i <= j:
            hi, hj = height[i], height[j]
            ans = max(ans, (j - i) * min(hi, hj))
            if hi >= hj:
                j -= 1
            else:
                i += 1
        
        return ans

File: Leet11.java
class Solution {
    public int maxArea(int[] height) {
        int len = height.length;
        int left = 0;
        int right = len-1;
        int max = Integer.MIN_VALUE;
        while (left < right) {
            max = Math.max(max, (right - left)*(Math.min(height[left], height[right])));
            if (height[left] <= height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }
}